advanced/Scripts/webpage.sh

```diff
@@ -147,67 +147,67 @@ SetWebPassword() {
 ProcessDNSSettings() {
     source "${setupVars}"
 
-    delete_dnsmasq_setting "server"
+    removeKey "${dnsmasqconfig}" "server"
 
     COUNTER=1
     while true ; do
         var=PIHOLE_DNS_${COUNTER}
         if [ -z "${!var}" ]; then
             break;
         fi
-        add_dnsmasq_setting "server" "${!var}"
+        addOrEditKeyValPair "${dnsmasqconfig}" "server" "${!var}"
```

Comment:
I think this is not going to work. If I understand the whole part right, it is reading `PIHOLE_DNS_X` from `setupVars` and adding it to `01-pi-hole.conf`. 
`addOrEditKeyValPair` is preventing adding multiple `server` lines, as it changes them always. You need

```suggestion
        addKey "${dnsmasqconfig}" "server=${!var}"
```

Reply:
Ah yes. That is a nice point. Currently mobile, so difficult to look - but do existing values get cleared out first, before we loop through PIHOLE_DNS_ values?

Reply:
Yes.

https://github.com/pi-hole/pi-hole/blob/23e6fa1ec56e7e24054d359ef5da0114e2f9b77f/advanced/Scripts/webpage.sh#L147-L154